testing/distributions-integ-tests/src/integTest/groovy/org/gradle/DistributionIntegrationSpec.groovy

```diff
@@ -167,7 +244,21 @@ abstract class DistributionIntegrationSpec extends AbstractIntegrationSpec {
         def coreLibs = contentsDir.file("lib").listFiles().findAll {
             it.name.startsWith("gradle-") && !it.name.startsWith("gradle-api-metadata") && !it.name.startsWith("gradle-kotlin-dsl")
         }
-        assert coreLibs.size() == coreLibJarsCount
+
+
+        def prefixedCoreLibNames = coreLibsFileNames.collect { "gradle-$it" }
+        def expectedCoreLibs = prefixedCoreLibNames.toSet()
+        def actualCoreLibs = coreLibs.collect { it.name - "${baseVersion}.jar" }.toSet()
+
+        def unknownCoreLibs = (actualCoreLibs - expectedCoreLibs).sort()
+        def missingCoreLibs = (expectedCoreLibs - actualCoreLibs).sort()
+
+        verifyAll {
+            unknownCoreLibs == []
+            missingCoreLibs == []
+        }
+
+        assert (prefixedCoreLibNames.clone().sort() == prefixedCoreLibNames) : "coreLibsFileNames has be sorted"
```

Comment:
💅  Instead of cloning, let's use `toSorted()` and don't worry about sorting anything in-place.
I would also apply that to the `unknownCoreLibs` and `missingCoreLibs` computations

Reply:
if I sort it in-place the comparison is always true, I checked :)

Reply:
I still think in-place sorts should be avoided unless required, and combining them with a `clone()` is a code smell